File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculo/inss/sobresalarios/TotalizadorInssSobreSalarios.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios;

import br.jus.trt8.pjecalc.base.comum.Utils;
import br.jus.trt8.pjecalc.negocio.comum.Total;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios.InssSobreSalarios;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios.OcorrenciaDeInss;
import java.math.BigDecimal;

public class TotalizadorInssSobreSalarios {
    private InssSobreSalarios inssSobreSalarios;
    private boolean isCalculado;
    private Total seguradoReclamadoCorrigido;
    private Total jurosSeguradoReclamadoCorrigido;
    private Total multaSeguradoReclamadoCorrigido;
    private Total totalSeguradoReclamadoCorrigido;
    private Total empresaFinalCorrigido;
    private Total jurosEmpresaFinalCorrigido;
    private Total multaEmpresaFinalCorrigido;
    private Total totalEmpresaFinalCorrigido;
    private Total satCorrigido;
    private Total jurosSatCorrigido;
    private Total multaSatCorrigido;
    private Total totalSatCorrigido;
    private Total terceirosCorrigido;
    private Total jurosTerceirosCorrigido;
    private Total multaTerceirosCorrigido;
    private Total totalTerceirosCorrigido;

    public TotalizadorInssSobreSalarios(InssSobreSalarios inssSobreSalarios) {
        this.inssSobreSalarios = inssSobreSalarios;
        this.reset();
    }

    public void reset() {
        this.isCalculado = false;
    }

    private TotalizadorInssSobreSalarios calcular() {
        if (!this.isCalculado) {
            this.seguradoReclamadoCorrigido = Total.newInstance(true);
            this.jurosSeguradoReclamadoCorrigido = Total.newInstance(true);
            this.multaSeguradoReclamadoCorrigido = Total.newInstance(true);
            this.totalSeguradoReclamadoCorrigido = Total.newInstance(true);
            this.empresaFinalCorrigido = Total.newInstance(true);
            this.jurosEmpresaFinalCorrigido = Total.newInstance(true);
            this.multaEmpresaFinalCorrigido = Total.newInstance(true);
            this.totalEmpresaFinalCorrigido = Total.newInstance(true);
            this.satCorrigido = Total.newInstance(true);
            this.jurosSatCorrigido = Total.newInstance(true);
            this.multaSatCorrigido = Total.newInstance(true);
            this.totalSatCorrigido = Total.newInstance(true);
            this.terceirosCorrigido = Total.newInstance(true);
            this.jurosTerceirosCorrigido = Total.newInstance(true);
            this.multaTerceirosCorrigido = Total.newInstance(true);
            this.totalTerceirosCorrigido = Total.newInstance(true);
            for (OcorrenciaDeInss ocorrenciaDeInss : this.inssSobreSalarios.getOcorrencias()) {
                if (ocorrenciaDeInss.isBaseVazia().booleanValue()) continue;
                if (Utils.naoNulo(ocorrenciaDeInss.getValorDevidoSeguradoFinalCorrigido())) {
                    this.seguradoReclamadoCorrigido.acumular(ocorrenciaDeInss.getValorDevidoSeguradoFinalCorrigido());
                }
                if (Utils.naoNulo(ocorrenciaDeInss.getJurosValorDevidoSeguradoFinal())) {
                    this.jurosSeguradoReclamadoCorrigido.acumular(ocorrenciaDeInss.getJurosValorDevidoSeguradoFinal());
                }
                if (Utils.naoNulo(ocorrenciaDeInss.getMultaValorDevidoSeguradoFinal())) {
                    this.multaSeguradoReclamadoCorrigido.acumular(ocorrenciaDeInss.getMultaValorDevidoSeguradoFinal());
                }
                if (Utils.naoNulo(ocorrenciaDeInss.getTotalValorDevidoSeguradoFinal())) {
                    this.totalSeguradoReclamadoCorrigido.acumular(ocorrenciaDeInss.getTotalValorDevidoSeguradoFinal());
                }
                if (Utils.naoNulo(ocorrenciaDeInss.getValorDevidoEmpresaFinalCorrigido())) {
                    this.empresaFinalCorrigido.acumular(ocorrenciaDeInss.getValorDevidoEmpresaFinalCorrigido());
                }
                if (Utils.naoNulo(ocorrenciaDeInss.getJurosValorDevidoEmpresaFinal())) {
                    this.jurosEmpresaFinalCorrigido.acumular(ocorrenciaDeInss.getJurosValorDevidoEmpresaFinal());
                }
                if (Utils.naoNulo(ocorrenciaDeInss.getMultaValorDevidoEmpresaFinal())) {
                    this.multaEmpresaFinalCorrigido.acumular(ocorrenciaDeInss.getMultaValorDevidoEmpresaFinal());
                }
                if (Utils.naoNulo(ocorrenciaDeInss.getTotalValorDevidoEmpresaFinal())) {
                    this.totalEmpresaFinalCorrigido.acumular(ocorrenciaDeInss.getTotalValorDevidoEmpresaFinal());
                }
                if (Utils.naoNulo(ocorrenciaDeInss.getValorDevidoSATCorrigido())) {
                    this.satCorrigido.acumular(ocorrenciaDeInss.getValorDevidoSATCorrigido());
                }
                if (Utils.naoNulo(ocorrenciaDeInss.getJurosValorDevidoSAT())) {
                    this.jurosSatCorrigido.acumular(ocorrenciaDeInss.getJurosValorDevidoSAT());
                }
                if (Utils.naoNulo(ocorrenciaDeInss.getMultaValorDevidoSAT())) {
                    this.multaSatCorrigido.acumular(ocorrenciaDeInss.getMultaValorDevidoSAT());
                }
                if (Utils.naoNulo(ocorrenciaDeInss.getTotalValorDevidoSAT())) {
                    this.totalSatCorrigido.acumular(ocorrenciaDeInss.getTotalValorDevidoSAT());
                }
                if (Utils.naoNulo(ocorrenciaDeInss.getValorDevidoTerceirosCorrigido())) {
                    this.terceirosCorrigido.acumular(ocorrenciaDeInss.getValorDevidoTerceirosCorrigido());
                }
                if (Utils.naoNulo(ocorrenciaDeInss.getJurosValorDevidoTerceiros())) {
                    this.jurosTerceirosCorrigido.acumular(ocorrenciaDeInss.getJurosValorDevidoTerceiros());
                }
                if (Utils.naoNulo(ocorrenciaDeInss.getMultaValorDevidoTerceiros())) {
                    this.multaTerceirosCorrigido.acumular(ocorrenciaDeInss.getMultaValorDevidoTerceiros());
                }
                if (!Utils.naoNulo(ocorrenciaDeInss.getTotalValorDevidoTerceiros())) continue;
                this.totalTerceirosCorrigido.acumular(ocorrenciaDeInss.getTotalValorDevidoTerceiros());
            }
            this.isCalculado = true;
        }
        return this;
    }

    public BigDecimal getSeguradoReclamadoCorrigido() {
        return Utils.arredondarValorMonetario(this.calcular().seguradoReclamadoCorrigido.getValor());
    }

    public BigDecimal getJurosSeguradoReclamadoCorrigido() {
        return Utils.arredondarValorMonetario(this.calcular().jurosSeguradoReclamadoCorrigido.getValor());
    }

    public BigDecimal getMultaSeguradoReclamadoCorrigido() {
        return Utils.arredondarValorMonetario(this.calcular().multaSeguradoReclamadoCorrigido.getValor());
    }

    public BigDecimal getTotalSeguradoReclamadoCorrigido() {
        return Utils.arredondarValorMonetario(this.calcular().totalSeguradoReclamadoCorrigido.getValor());
    }

    public BigDecimal getEmpresaFinalCorrigido() {
        return Utils.arredondarValorMonetario(this.calcular().empresaFinalCorrigido.getValor());
    }

    public BigDecimal getJurosEmpresaFinalCorrigido() {
        return Utils.arredondarValorMonetario(this.calcular().jurosEmpresaFinalCorrigido.getValor());
    }

    public BigDecimal getMultaEmpresaFinalCorrigido() {
        return Utils.arredondarValorMonetario(this.calcular().multaEmpresaFinalCorrigido.getValor());
    }

    public BigDecimal getTotalEmpresaFinalCorrigido() {
        return Utils.arredondarValorMonetario(this.calcular().totalEmpresaFinalCorrigido.getValor());
    }

    public BigDecimal getSatCorrigido() {
        return Utils.arredondarValorMonetario(this.calcular().satCorrigido.getValor());
    }

    public BigDecimal getJurosSatCorrigido() {
        return Utils.arredondarValorMonetario(this.calcular().jurosSatCorrigido.getValor());
    }

    public BigDecimal getMultaSatCorrigido() {
        return Utils.arredondarValorMonetario(this.calcular().multaSatCorrigido.getValor());
    }

    public BigDecimal getTotalSatCorrigido() {
        return Utils.arredondarValorMonetario(this.calcular().totalSatCorrigido.getValor());
    }

    public BigDecimal getTerceirosCorrigido() {
        return Utils.arredondarValorMonetario(this.calcular().terceirosCorrigido.getValor());
    }

    public BigDecimal getJurosTerceirosCorrigido() {
        return Utils.arredondarValorMonetario(this.calcular().jurosTerceirosCorrigido.getValor());
    }

    public BigDecimal getMultaTerceirosCorrigido() {
        return Utils.arredondarValorMonetario(this.calcular().multaTerceirosCorrigido.getValor());
    }

    public BigDecimal getTotalTerceirosCorrigido() {
        return Utils.arredondarValorMonetario(this.calcular().totalTerceirosCorrigido.getValor());
    }
}

File: src/lib/pjecalc/core/dominio/calculo/inss/sobresalarios/totalizador-inss-sobre-salarios.ts
/**
 * PJe-Calc v2.15.1 — TotalizadorInssSobreSalarios
 * Porte 1:1 de: br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios.TotalizadorInssSobreSalarios
 *
 * Ref Java: pjecalc-fonte/.../sobresalarios/TotalizadorInssSobreSalarios.java
 *
 * Acumula 16 totais (4 grupos × 4 categorias) a partir das ocorrências do
 * InssSobreSalarios referenciado. `calcular()` é lazy (roda uma vez e guarda
 * `isCalculado`). `reset()` invalida o cache para recálculo.
 */
import Decimal from 'decimal.js';
import { arredondarValorMonetario } from '../../../../base/comum/utils';
import type { InssSobreSalarios } from './inss-sobre-salarios';
import type { OcorrenciaDeInss } from './ocorrencia-de-inss';

const ZERO = new Decimal(0);

/** Mimetiza `br.jus.trt8.pjecalc.negocio.comum.Total` (acumulador Decimal) */
class Total {
  private valor: Decimal = ZERO;
  acumular(v: Decimal | null): void { if (v !== null) this.valor = this.valor.plus(v); }
  getValor(): Decimal { return this.valor; }
}

export class TotalizadorInssSobreSalarios {
  private inssSobreSalarios: InssSobreSalarios;
  private isCalculado = false;
  private seguradoReclamadoCorrigido = new Total();
  private jurosSeguradoReclamadoCorrigido = new Total();
  private multaSeguradoReclamadoCorrigido = new Total();
  private totalSeguradoReclamadoCorrigido = new Total();
  private empresaFinalCorrigido = new Total();
  private jurosEmpresaFinalCorrigido = new Total();
  private multaEmpresaFinalCorrigido = new Total();
  private totalEmpresaFinalCorrigido = new Total();
  private satCorrigido = new Total();
  private jurosSatCorrigido = new Total();
  private multaSatCorrigido = new Total();
  private totalSatCorrigido = new Total();
  private terceirosCorrigido = new Total();
  private jurosTerceirosCorrigido = new Total();
  private multaTerceirosCorrigido = new Total();
  private totalTerceirosCorrigido = new Total();

  constructor(inssSobreSalarios: InssSobreSalarios) {
    this.inssSobreSalarios = inssSobreSalarios;
  }

  reset(): void { this.isCalculado = false; }

  private calcular(): this {
    if (this.isCalculado) return this;
    // Reinicia totais
    this.seguradoReclamadoCorrigido = new Total();
    this.jurosSeguradoReclamadoCorrigido = new Total();
    this.multaSeguradoReclamadoCorrigido = new Total();
    this.totalSeguradoReclamadoCorrigido = new Total();
    this.empresaFinalCorrigido = new Total();
    this.jurosEmpresaFinalCorrigido = new Total();
    this.multaEmpresaFinalCorrigido = new Total();
    this.totalEmpresaFinalCorrigido = new Total();
    this.satCorrigido = new Total();
    this.jurosSatCorrigido = new Total();
    this.multaSatCorrigido = new Total();
    this.totalSatCorrigido = new Total();
    this.terceirosCorrigido = new Total();
    this.jurosTerceirosCorrigido = new Total();
    this.multaTerceirosCorrigido = new Total();
    this.totalTerceirosCorrigido = new Total();

    const ocorrencias = this.inssSobreSalarios.getOcorrencias();
    const iter: Iterable<OcorrenciaDeInss> = Array.isArray(ocorrencias)
      ? ocorrencias
      : (ocorrencias as Set<OcorrenciaDeInss>);
    for (const oc of iter) {
      if (oc.isBaseVazia()) continue;
      this.seguradoReclamadoCorrigido.acumular(oc.getValorDevidoSeguradoFinalCorrigido());
      this.jurosSeguradoReclamadoCorrigido.acumular(oc.getJurosValorDevidoSeguradoFinal());
      this.multaSeguradoReclamadoCorrigido.acumular(oc.getMultaValorDevidoSeguradoFinal());
      this.totalSeguradoReclamadoCorrigido.acumular(oc.getTotalValorDevidoSeguradoFinal());

      this.empresaFinalCorrigido.acumular(oc.getValorDevidoEmpresaFinalCorrigido());
      this.jurosEmpresaFinalCorrigido.acumular(oc.getJurosValorDevidoEmpresaFinal());
      this.multaEmpresaFinalCorrigido.acumular(oc.getMultaValorDevidoEmpresaFinal());
      this.totalEmpresaFinalCorrigido.acumular(oc.getTotalValorDevidoEmpresaFinal());

      this.satCorrigido.acumular(oc.getValorDevidoSATCorrigido());
      this.jurosSatCorrigido.acumular(oc.getJurosValorDevidoSAT());
      this.multaSatCorrigido.acumular(oc.getMultaValorDevidoSAT());
      this.totalSatCorrigido.acumular(oc.getTotalValorDevidoSAT());

      this.terceirosCorrigido.acumular(oc.getValorDevidoTerceirosCorrigido());
      this.jurosTerceirosCorrigido.acumular(oc.getJurosValorDevidoTerceiros());
      this.multaTerceirosCorrigido.acumular(oc.getMultaValorDevidoTerceiros());
      this.totalTerceirosCorrigido.acumular(oc.getTotalValorDevidoTerceiros());
    }
    this.isCalculado = true;
    return this;
  }

  getSeguradoReclamadoCorrigido(): Decimal { return arredondarValorMonetario(this.calcular().seguradoReclamadoCorrigido.getValor()); }
  getJurosSeguradoReclamadoCorrigido(): Decimal { return arredondarValorMonetario(this.calcular().jurosSeguradoReclamadoCorrigido.getValor()); }
  getMultaSeguradoReclamadoCorrigido(): Decimal { return arredondarValorMonetario(this.calcular().multaSeguradoReclamadoCorrigido.getValor()); }
  getTotalSeguradoReclamadoCorrigido(): Decimal { return arredondarValorMonetario(this.calcular().totalSeguradoReclamadoCorrigido.getValor()); }

  getEmpresaFinalCorrigido(): Decimal { return arredondarValorMonetario(this.calcular().empresaFinalCorrigido.getValor()); }
  getJurosEmpresaFinalCorrigido(): Decimal { return arredondarValorMonetario(this.calcular().jurosEmpresaFinalCorrigido.getValor()); }
  getMultaEmpresaFinalCorrigido(): Decimal { return arredondarValorMonetario(this.calcular().multaEmpresaFinalCorrigido.getValor()); }
  getTotalEmpresaFinalCorrigido(): Decimal { return arredondarValorMonetario(this.calcular().totalEmpresaFinalCorrigido.getValor()); }

  getSatCorrigido(): Decimal { return arredondarValorMonetario(this.calcular().satCorrigido.getValor()); }
  getJurosSatCorrigido(): Decimal { return arredondarValorMonetario(this.calcular().jurosSatCorrigido.getValor()); }
  getMultaSatCorrigido(): Decimal { return arredondarValorMonetario(this.calcular().multaSatCorrigido.getValor()); }
  getTotalSatCorrigido(): Decimal { return arredondarValorMonetario(this.calcular().totalSatCorrigido.getValor()); }

  getTerceirosCorrigido(): Decimal { return arredondarValorMonetario(this.calcular().terceirosCorrigido.getValor()); }
  getJurosTerceirosCorrigido(): Decimal { return arredondarValorMonetario(this.calcular().jurosTerceirosCorrigido.getValor()); }
  getMultaTerceirosCorrigido(): Decimal { return arredondarValorMonetario(this.calcular().multaTerceirosCorrigido.getValor()); }
  getTotalTerceirosCorrigido(): Decimal { return arredondarValorMonetario(this.calcular().totalTerceirosCorrigido.getValor()); }
}
